File: src/TinyBlog.DataServices/Entities/Layout.cs
﻿using MongoDB.Bson.Serialization.Attributes;

namespace TinyBlog.DataServices.Entities
{
    [BsonIgnoreExtraElements]
    public class Layout : BaseEntity
    {
        [BsonElement("title")]
        public string Title { get; set; }

        [BsonElement("description")]
        public string Description { get; set; }

        [BsonElement("uri")]
        public string Uri { get; set; }

        [BsonElement("author")]
        public string Author { get; set; }

        [BsonElement("language")]
        public string Language { get; set; }

        [BsonElement("googleTagsCode")]
        public string GoogleTagsCode { get; set; }

        [BsonElement("headerContent")]
        public string HeaderContent { get; set; }

        [BsonElement("footerContent")]
        public string FooterContent { get; set; }
    }
}

File: src/TinyBlog.Web/Scripts/testcafe/types/layout.ts
import * as mongoose from 'mongoose';

export interface LayoutDomain extends mongoose.Document {
    title: string;
    description: string;
    uri: string;
    author: string;
    language: string;
    googleTagsCode: string;
    headerContent: string;
    footerContent: string;
}

const LayoutSchema: mongoose.Schema = new mongoose.Schema({
    title: { type: String },
    description: { type: String },
    uri: { type: String },
    author: { type: String },
    language: { type: String },
    googleTagsCode: { type: String },
    headerContent: { type: String },
    footerContent: { type: String }
});

export const Layout = mongoose.model<LayoutDomain>('layout', LayoutSchema, 'layout');